components/widgets/menu/menu_items/submenu_item.jsx

```diff
@@ -0,0 +1,168 @@
+            event.nativeEvent.path.slice(0, 2).find((e) => e.id === id)) {
+            action(id);
+        } else if (!event.nativeEvent.path) { //for tests only that don't contain `path`
+            action(id);
+        }
+    }
+
+    render() {
+        const {id, text, subMenu, icon, action, xOffset, ariaLabel} = this.props;
+        const isMobile = Utils.isMobile();
+
+        let textProp = text;
+        if (icon) {
+            textProp = (
+                <React.Fragment>
+                    <span className='icon'>{icon}</span>
+                    {text}
+                </React.Fragment>
+            );
+        }
+
+        const hasSubmenu = subMenu && subMenu.length;
+        const parentWidth = this.node && this.node.current ? this.node.current.getBoundingClientRect().width - 19 : 0;
+        const subMenuStyle = {
+            visibility: this.state.show && hasSubmenu && !isMobile ? 'visible' : 'hidden',
+            right: (parseInt(xOffset, 10) - 1) + 'px',
+        };
+
+        let subMenuContent = '';
+
+        if (!isMobile) {
+            subMenuContent = (
+                <ul
+                    className={'a11y__popup Menu dropdown-menu SubMenu'}
+                    style={subMenuStyle}
+                >
+                    {isMobile || !subMenu ? '' : subMenu.map((s) => {
+                        return (
+                            <SubMenuItem
+                                key={s.id}
+                                id={s.id}
+                                text={s.text}
+                                subMenu={s.subMenu}
+                                action={action}
+                                xOffset={parentWidth}
+                                aria-label={ariaLabel}
+                                root={false}
+                            />
+                        );
+                    })}
+                </ul>
+            );
+        }
+
+        return (
+            <li
+                className='MenuItem'
+                role='menuitem'
+                id={id + '_menuitem'}
+                ref={this.node}
+            >
+                <div
+                    id={id}
+                    aria-label={ariaLabel}
+                    onMouseEnter={this.show}
+                    onMouseLeave={this.hide}
+                    onClick={this.onClick}
+                >
+                    <span
+                        id={'channelHeaderDropdownIconLeft_' + id}
+                        className={'fa fa-angle-left SubMenu__icon-left' + (hasSubmenu && !isMobile ? '' : '-empty' + (isMobile ? ' Mobile' : ''))}
+                        aria-label='submenu icon'
+                    />
+                    {textProp}
+                    <span
+                        id={'channelHeaderDropdownIconRight_' + id}
+                        className={'fa fa-angle-right SubMenu__icon-right' + (hasSubmenu && isMobile ? '' : '-empty')}
+                        aria-label='submenu icon'
```

Comment:
I think the aria-label should be translated.

Reply:
done